textacy/datasets/wikipedia.py

```diff
@@ -146,23 +153,34 @@ class Wikipedia(Dataset):
             most recently available version or a date formatted as "YYYYMMDD".
             Dumps are produced intermittently; check for available versions at
             https://meta.wikimedia.org/wiki/Data_dumps.
-
+        project (str): Name or alias of wikimedia data dump you wish to work
+           with.  Currently 'wiki' (default) and 'wikinews' are supported with
+           respective aliases 'wikipedia' and 'news'.
     Attributes:
         lang (str): Standard two-letter language code used in instantiation.
         version (str): Database dump version used in instantiation.
         filestub (str): The component of ``filename`` that is unique to this
             lang- and version-specific database dump.
         filename (str): Full path on disk for the lang- and version-specific
             Wikipedia database dump, found under the ``data_dir`` directory.
+        project (str): the canonical name for the wikimedia project which is
+            used as a component of ``filestub``.
     """
 
-    def __init__(self, data_dir=DATA_DIR, lang='en', version='latest'):
+    def __init__(self, data_dir=DATA_DIR, lang='en', version='latest', project='wiki'):
         super(Wikipedia, self).__init__(
             name=NAME, description=DESCRIPTION, site_url=SITE_URL, data_dir=data_dir)
         self.lang = lang
         self.version = version
-        self.filestub = '{lang}wiki/{version}/{lang}wiki-{version}-pages-articles.xml.bz2'.format(
-            version=self.version, lang=self.lang)
+        proj = PROJECTS.get(project, None)
+        if proj is None:
+            raise AttributeError(
+                'Invalid or unsupported wikimedia project: {project}'.format(
+                project=project
+            ))
+        self.project = proj
+        self.filestub = '{lang}{project}/{version}/{lang}{project}-{version}-pages-articles.xml.bz2'.format(
```

Comment:
I did a spot-check, and it looks like this filestub format still works for wikinews in various languages. I'm thinking the "check URL via HEAD request" idea you mentioned is probably worth doing, but that can be added separately.

Reply:
I'll add this feature (as well as the other changes you suggested above) in a commit.  This will be more robust than merely relying on both my/your spot-checking of the URLs it produces as it can return an error to the user in the event of a HEAD 404. In the event of a 404 I'll additionally backoff from the data dump file URL to the {lang}{project} root URL and verify that the project for that language exists and report a different message if it doesn't.